src/commands/CreateDataSourceEntryCommand.ts

```diff
@@ -14,51 +14,97 @@
+			vscode.window.showInformationMessage(`New datasource entry ${entryName} has been created successfully...`);
+		} else {
+			vscode.window.showErrorMessage(`An error occured when trying to create a new datasource entry ${entryName} in datasource ${dsNode.label}...`);
+		}
+	}
 }
 
-export function handleDataSourceEntryCreation(dvConfig: IDVConfig, dsConfig: IDataSourceConfig, file: string, entryName: string, entryValue: string): Promise<boolean> {
+export function handleDataSourceEntryCreation(dvConfig: DataVirtConfig, dsConfig: DataSourceConfig, file: string, entryType: string, entryName: string, entryValue: string, refName?: string, refKey?: string): Promise<boolean> {
 	return new Promise<boolean>( (resolve) => {
-		if (dvConfig && dsConfig && file && entryName && entryValue) {
+		if (dvConfig && entryType && dsConfig && file && entryName && entryValue) {
 			try {
-				if (!dsConfig.entries.has(entryName.toUpperCase())) {
-					dsConfig.entries.set(entryName.toUpperCase(), entryValue);
+				const entry: Property = utils.getDataSourceEntryByName(entryName, dsConfig);
+				if (!entry) {
+					setDataSourceEntryValue(dsConfig.properties, entryType, entryName, entryValue, refName, refKey);
+					createOrUpdateLocalReferenceFile(refName, refKey, entryValue, entryType);
 				} else {
 					resolve(false);
 				}
-				utils.mapDSConfigToEnv(dsConfig, dvConfig);
 				utils.saveModelToFile(dvConfig, file);
 				resolve(true);
 			} catch (error) {
 				extension.log(error);
 				resolve(false);
 			}
 		} else {
-			extension.log('handleDataSourceEntryCreation: Unable to create the datasource entry...');
+			extension.log(`handleDataSourceEntryCreation: Unable to create the datasource entry ${entryName ? entryName : '<Unknown>'} in datasource ${dsConfig ? dsConfig.name : '<Unknown>'}...`);
 			resolve(false);
 		}
 	});
 }
+
+function setDataSourceEntryValue(entries: Property[], entryType: string, entryName: string, entryValue: string, refName?: string, refKey?: string): void {
+	let entry: Property;
+	if (entryType === constants.DATASOURCE_ENTRY_TYPE_SECRET) {
+		const secretRef = new SecretRef(new KeyRef(refName, refKey));
+		const value = new ValueFrom(secretRef);
+		entry = new Property(entryName, undefined, value);
+	} else if (entryType === constants.DATASOURCE_ENTRY_TYPE_CONFIGMAP) {
+		const configMapRef = new ConfigMapRef(new KeyRef(refName, refKey));
+		const value = new ValueFrom(configMapRef);
+		entry = new Property(entryName, undefined, value);
+	} else {
+		entry = new Property(entryName, entryValue, undefined);
+	}
+	entries.push(entry);
+}
+
+function createOrUpdateLocalReferenceFile(refName: string, refKey: string, entryValue: string, entryType: string) {
+	// TODO: create or update the entry reference in a local yaml file for configMap OR secret format
+	// TODO: implement me!
+	extension.log(`call to unimplemented function createOrUpdateLocalReferenceFile(${refName}, ${refKey}, ${entryValue}, ${entryType})...`);
```

Comment:
I cannot find a JIRA issue for it https://issues.redhat.com/issues/?jql=project%20%3D%20FUSETOOLS2%20AND%20text%20~%20createOrUpdateLocalReferenceFile

@lhein can you point to it? si it marked as blocker?

seems like a must to be implemented before a first release (or to remove code related to that)

Reply:
https://issues.redhat.com/browse/FUSETOOLS2-365

And you are right. That comment needs to be disabled before the release.